#24308 Broke python3 interpreter selection

**Describe the bug**
https://github.com/zephyrproject-rtos/zephyr/pull/24308 broke selection of the Python3 interpreter. Instead of using system (and user's) default "python3" interpreter, it picks up random binaries in uncalled-for manner.

**To Reproduce**
Since #24308 was merged, any attempt to build any Zephyr application leads to error below.
Specific steps to reproduce the behavior:
1. cd samples/net/sockets/big_http_download
2. `west build -b frdm_k64f -- -DOVERLAY_CONFIG=overlay-ci.conf`
3. See error
~~~
$ west build -b frdm_k64f -- -DOVERLAY_CONFIG=overlay-ci.conf
-- west build: generating a build system
Including boilerplate (Zephyr base): /home/pfalcon/projects-3rdparty/Embedded/Zephyr/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/pfalcon/projects-3rdparty/Embedded/Zephyr/zephyr/samples/net/sockets/big_http_download
-- Zephyr version: 2.2.99 (/home/pfalcon/projects-3rdparty/Embedded/Zephyr/zephyr)
-- Found Python3: /usr/bin/python3.8 (found suitable version "3.8.2", minimum required is "3.6") found components:  Interpreter 
-- Board: frdm_k64f
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'west'
CMake Error at /home/pfalcon/projects-3rdparty/Embedded/Zephyr/zephyr/cmake/host-tools.cmake:26 (message):
  Unable to import west.version from '/usr/bin/python3.8'
Call Stack (most recent call first):
  /home/pfalcon/projects-3rdparty/Embedded/Zephyr/zephyr/cmake/app/boilerplate.cmake:483 (include)
  /home/pfalcon/projects-3rdparty/Embedded/Zephyr/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/pfalcon/projects-3rdparty/Embedded/Zephyr/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /home/pfalcon/.local/bin/cmake -B/mnt/hdd/projects-3rdparty/Embedded/Zephyr/zephyr/samples/net/sockets/big_http_download/build -S/mnt/hdd/projects-3rdparty/Embedded/Zephyr/zephyr/samples/net/sockets/big_http_download -GNinja -DBOARD=frdm_k64f -DOVERLAY_CONFIG=overlay-ci.conf
~~~

This happens because:
1. I have a system-default Python interpreter installed as `python3` executable, per the best practices. I'm running Ubuntu 18.04, so this happens to be Pyhon3.6.
2. As a regular Python user, I have many other Python versions installed, e.g. Python3.5, Python3.7, Python3.8, Python3.9, Python-git, etc. etc.

Per the common sense and the best practices, I expect any software which requires Python3 to use system-default executable (`python3`), unless I explicitly used a different Python interpreter to run a specific software. 

**Expected behavior**
Zephyr build system should not try to oversmart user and select arbitrary binaries to run behind user's back. For as long as system-default `python3` interpreter exists, it should use that, unless explicitly asked by a user to do otherwise (via command-line arg or environment var).

**Impact**
This issue breaks development process for any user who may have a newer version of Python installed for reasons not related to Zephyr.

**Environment (please complete the following information):**
 - OS: Linux, Ubuntu 18.04
 - Reproduced with master 4215968b74666adedc9c0d643b38d0c874319512

--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -9,5 +9,29 @@ if (WIN32)
   set(ENV{PYTHONIOENCODING} "utf-8")
 endif()
 
-find_package(Python3 3.6 REQUIRED)
+set(PYTHON_MINIMUM_REQUIRED 3.6)
+
+# We are using foreach here, instead of find_program(PYTHON_EXECUTABLE_SYSTEM_DEFAULT "python" "python3")
+# cause just using find_program directly could result in a python2.7 as python, and not finding a valid python3.
+foreach(PYTHON_PREFER ${PYTHON_PREFER} "python" "python3")
+  find_program(PYTHON_PREFER_EXECUTABLE ${PYTHON_PREFER})
+  if(PYTHON_PREFER_EXECUTABLE)
+      execute_process (COMMAND "${PYTHON_PREFER_EXECUTABLE}" -c
+                               "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"
+                       RESULT_VARIABLE result
+                       OUTPUT_VARIABLE version
+                       ERROR_QUIET
+                       OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+     if(version VERSION_LESS PYTHON_MINIMUM_REQUIRED)
+       set_property (CACHE PYTHON_PREFER_EXECUTABLE PROPERTY VALUE "PYTHON_PREFER_EXECUTABLE-NOTFOUND")
+     else()
+       set(PYTHON_MINIMUM_REQUIRED ${version})
+       set(PYTHON_EXACT EXACT)
+       break()
+     endif()
+  endif()
+endforeach()
+
+find_package(Python3 ${PYTHON_MINIMUM_REQUIRED} REQUIRED ${PYTHON_EXACT})
 set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})

